Makefile

```
@@ -1,2 +1,22 @@
-build:
+.PHONY: build
+build: ## Compiles CLI binary.
 	@go build -o ./fleetlockctl main.go
+
+.PHONY: test-working-tree-clean
+test-working-tree-clean: ## Checks if working directory is clean.
+	@test -z "$$(git status --porcelain)" || (echo "Commit all changes before running this target"; exit 1)
+
+.PHONY: update-linters
+update-linters: ## Updates list of linters in .golangci.yml file based on currently installed golangci-lint binary.
```

so this, is something I seen sometimes in order to pull the "dependencies" required by the linters, etc.

```Makefile
TOOL_BIN := $(PWD)/bin
GOLINT := $(TOOL_BIN)/golangci-lint
...
$(GOLINT):
	@GOBIN=$(TOOL_BIN) go install .../golanci-lint
        # or other way to install the tool
```

Then, we can the tools as dependency of the target:
```suggestion
update-linters: $(GOLINT) ## Updates list of linters in .golangci.yml file based on currently installed golangci-lint binary.
```

Of course, the `TOOL_BIN` is gitignored.

---

IMO starting with Go 1.17, the best way is to use `go run`, which does not touch the `go.mod` file anymore, so if you're fine with it, I'd recommend using that.

---

Done. Please have a look :)

---

> which does not touch the go.mod file anymore

happy to learn that !